x-pack/plugins/security_solution/public/common/components/guided_onboarding_tour/tour.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ReactChild } from 'react';
+import React, { createContext, useCallback, useContext, useEffect, useMemo, useState } from 'react';
+
+import useObservable from 'react-use/lib/useObservable';
+import { catchError, of, timeout } from 'rxjs';
+import { useKibana } from '../../lib/kibana';
+import { securityTourConfig, SecurityStepId } from './tour_config';
+
+export interface TourContextValue {
+  activeStep: number;
+  endTourStep: (stepId: SecurityStepId) => void;
+  incrementStep: (stepId: SecurityStepId, step?: number) => void;
+  isTourShown: (stepId: SecurityStepId) => boolean;
+}
+
+const initialState: TourContextValue = {
+  activeStep: 0,
+  endTourStep: () => {},
+  incrementStep: () => {},
+  isTourShown: () => false,
+};
+
+const TourContext = createContext<TourContextValue>(initialState);
+
+export const TourContextProvider = ({ children }: { children: ReactChild }) => {
+  const { guidedOnboardingApi } = useKibana().services.guidedOnboarding;
+
+  const isRulesTourActive = useObservable(
+    guidedOnboardingApi?.isGuideStepActive$('security', SecurityStepId.rules).pipe(
+      // if no result after 30s the observable will error, but the error handler will just emit false
+      timeout(30000),
+      catchError((error) => of(false))
+    ) ?? of(false),
+    false
+  );
+  const isAlertsCasesTourActive = useObservable(
+    guidedOnboardingApi?.isGuideStepActive$('security', SecurityStepId.alertsCases).pipe(
+      // if no result after 30s the observable will error, but the error handler will just emit false
+      timeout(30000),
+      catchError((error) => of(false))
+    ) ?? of(false),
+    false
+  );
+
+  const tourStatus = useMemo(
+    () => ({
+      [SecurityStepId.rules]: isRulesTourActive,
+      [SecurityStepId.alertsCases]: isAlertsCasesTourActive,
+    }),
+    [isRulesTourActive, isAlertsCasesTourActive]
+  );
+
+  const isTourShown = useCallback((stepId: SecurityStepId) => tourStatus[stepId], [tourStatus]);
+  const [activeStep, _setActiveStep] = useState<number>(1);
+
+  const incrementStep = useCallback((stepId: SecurityStepId, step?: number) => {
+    _setActiveStep((prevState) =>
+      step != null && step <= securityTourConfig[stepId].length
+        ? step
+        : (prevState >= securityTourConfig[stepId].length ? 0 : prevState) + 1
+    );
+  }, []);
+
+  const resetStep = useCallback(() => {
+    _setActiveStep(1);
+  }, []);
+
+  useEffect(() => {
+    resetStep();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why is this needed?

Reply:
idk that it is... put it in for safety but now im looking at it and its a little dumb lol. lets delete it!

Reply:
ok thinking further... i think i put it in because idk how the transition between the global steps will be. maybe i will remove it in future PR if it is not needed